### ISSUE TITLE
Unable to find Google Chrome

### ISSUE THREAD

**Issue by TraderStf** [dmarmor/epichrome#28]
Hello,

I have tried to create several, local or external disks, various folders or name...
Always the same error.

EC 2.15 - OS X 10.10.5 French

Thanks,

Ready to create!
App: My Epichrome App.app
Menubar Name: MeEpichrome
Path: /Volumes/GDISK/zEpichrome
Style: App Window
URL: https://www.google.com/mail/				
Register as Browser: No
Icon: <default>

Creation failed: Unable to find Google Chrome application. (return POSIX path of (choose application with title "Locate Chrome" with prompt "Please locate Google Chrome" as alias))

**Comment by dmarmor** [comment]
Sorry you're having trouble...can you tell me where you have Chrome installed on your system? Also would be helpful if you could send me the output of this terminal command:

```bash
mdfind "kMDItemCFBundleIdentifier == 'com.google.Chrome'"
```

Thanks!

**Comment by TraderStf** [comment]
No problemo :-)
It's in the usual Applications folder.
The mdfind returns nothing, no matter from where / or ~, even with a sudo.
I search with FindAnyFile as admin, no duplicate of GC.
GC works and updates without problem.
Safari is my default browser.
I have fixed HD authorizations.

I will try to find out what might be the problem.
Thank you very much!

**Comment by dmarmor** [comment]
Hmm, yeah, that's weird. That command should give you something that looks like:

```bash
prompt> mdfind "kMDItemCFBundleIdentifier == 'com.google.Chrome'"
/Applications/Google Chrome.app
```

The fact that it's not is why creation is failing, but I really can't figure out why it would fail if Chrome is installed on your system. It's returning no text at all, huh? Can you try this for me:

```bash
mdfind "kMDItemCFBundleIdentifier == 'com.google.Chrome'"
echo result is "$?"
```

That'll tell us if mdfind is returning any kind of error code. Another (unlikely) problem could be that it's somehow not the right mdfind. Maybe try this:

```bash
mdfind -h
```

That should print a usage message that looks like this:

```bash
Usage: mdfind [-live] [-count] [-onlyin directory] [-name fileName | -s smartFolderName | query]
list the files matching the query
query can be an expression or a sequence of words

	-attr <attr>      Fetches the value of the specified attribute
	-count            Query only reports matching items count
	-onlyin <dir>     Search only within given directory
	-live             Query should stay active
	-name <name>      Search on file name only
	-reprint          Reprint results on live update
	-s <name>         Show contents of smart folder <name>
	-0                Use NUL (``\0'') as a path separator, for use with xargs -0.

example:  mdfind image
example:  mdfind -onlyin ~ image
example:  mdfind -name stdlib.h
example:  mdfind "kMDItemAuthor == '*MyFavoriteAuthor*'"
example:  mdfind -live MyFavoriteAuthor
```

Let me know what you get back from any of those tests? Thanks!

**Comment by TraderStf** [comment]
Result is 0 (a zero)

I have tried, other apps TextEdit, BBEdit... from other path when typing in terminal... zero

Yesterday, I did this procedure, just in case some permissions/chmod were incorrect, afterwards can not install, always same message.
I did that because some "Save as" were refused in some 'normal' folders.

<pre>
At startup chime, immediately hold down Command-R to boot into Recovery mode.
Select the menu item Utilties > Terminal.
Type resetpassword and press Return.
Don’t worry: this won’t reset your password.
Pick your username from the Select the User Account dropdown menu.
In the lower-right corner of the screen, click Reset next to the Reset Home Directory Permissions and ACLs area.
Restart
</pre>

Always zero
After that I use Onyx to clean almost everything, always zero.

Is mdfind linked/using Spotlight?
Because I have disabled spotlight if I remember well, first from its Privacy Panel, put all my HD.
Then with a cmd in terminal

BRB on sunday

**Comment by breeden** [comment]
If I understand correctly, mdfind is just a command line interface for
accessing Spotlight's drive index. So, if Spotlight is disabled completely,
then mdfind wouldn't have access to any of your applications, not just GC.

To test this, you can try using mdfind to search for any file whose
location you know. For instance, if I have an image on my desktop called
IMG_123.jpg, then I should have the following output:

>> mdfind IMG_123.jpg
/Users/katherine/Desktop/IMG_123.jpg

For you, if mdfind returns 0, that means it doesn't have access to a drive
index.

You may be able to manually reindex Spotlight from the command line without
actually enabling Spotlight. To do this for the whole drive, you would run:

>> sudo mdutil -E /

However, for these purposes you might be able to just manually add the
individual missing entry:

>> mdimport /Applications/Google\ Chrome.app

Maybe that will work!
KB

On Fri, Sep 18, 2015 at 11:37 PM, Stéphane Moureau <notifications@github.com

**Comment by TraderStf** [comment]
I have read this article to see if all files mentioned were in my mac.
http://recomhub.com/blog/how-to-completely-disable-spotlight-on-mac-os-x/
http://recomhub.com/blog/how-to-disable-or-enable-spotlight-in-mac-os-x-yosemite-mavericks-mountain-lion/

All were present but not "/etc/hostconfig"

I have
"host" with the 3 usual IP lines.
"hosts.equiv" is empty

In Onyx, I "Misc" Tab, first line, I activate the indexation of start up disk.

Now the mdfind are working and I was able to create the default "app" with epichrome.

Thanks a lot!

**Comment by breeden** [comment]
C'est chouette! Have fun epichrome-ing!
KB

On Mon, Sep 21, 2015 at 12:19 AM, Stéphane Moureau <notifications@github.com

**Comment by dmarmor** [comment]
Thanks KB! I confirmed that if Spotlight indexing is turned off (or Spotlight is disabled entirely), Epichrome was not a happy camper. And, it turns out I think this issue and another one I couldn't solve ([#23](https://github.com/dmarmor/epichrome/issues/23)) were actually the same issue!

Anyway, I've just released [version 2.1.6](https://github.com/dmarmor/epichrome/releases), which I *think* solves the problem. Please do read the release notes regarding updating your existing apps, and then give it a try. It should allow you to still create and use Epichrome apps even with Spotlight turned off (with a few limitations detailed in the Readme).

Anyway, let me know how it works! For now, I'll close the issue, but feel free to re-open if you're still having problems.
